specification/edgeorder/resource-manager/Microsoft.EdgeOrder/2020-12-01-preview/examples/OrderCollectionGet.json

```diff
@@ -0,0 +1,34 @@
+{
+  "parameters": {
+    "subscriptionId": "fa68082f-8ff7-4a25-95c7-ce9da541242f",
+    "resourceGroupName": "TestRG",
+    "orderCollectionName": "TestOrderCollectionName1",
+    "api-version": "2020-12-01-preview",
+    "User-Agent": [
+      "vscode-restclient"
+    ],
+    "Referer": [
+      "https://localhost/subscriptions/fa68082f-8ff7-4a25-95c7-ce9da541242f/resourceGroups/TestRG/providers/Microsoft.EdgeOrder/orderCollections/TestOrderCollectionName1?api-version=2020-12-01-preview"
+    ]
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "properties": {
+          "orderIds": [
+            "/subscriptions/fa68082f-8ff7-4a25-95c7-ce9da541242f/resourceGroups/TestRG/providers/Microsoft.EdgeOrder/orders/TestOrderName1",
```

Comment:
>TestOrderName1 [](start = 124, length = 14)

so from a customer experience in the portal - they will place an order - this will go and create the individual orders and then the order collection right? what if one of the order creation fails due to 4xx/5xx? The order collection will then only contain the successfully created order ids?

Reply:
Yes portal/client should create orders and then ordercollections.  Order collection creation will be failed even if one order in the collection fails. We are not supporting partial order scenario.

Reply:
But then you end up with orphaned orders right i.e. which have no corresponding order collection. Is that supported/ will that work?

Reply:
btw, since the order collection api and order api are different - you cannot ensure the atomicity of the operation you are talking about esp. if the client uses api/ps/cli

Reply:
Yes. If there is no ordercollection the orders will be orphaned. It wont make any progress in state transition. We ll cleanup the orphaned orders after some threshold period(yet to be decided)

Reply:
have you considered the design where the ordercollection is a parent resource of the order instead of sibling? so an order is ordercollections/{orderCollectionName}/orders/{orderName}?"

Reply:
Yes. We have a use case to support resource move at order level. It is quite complex to implement all validation ARM does today from our side